assignment3.txt

```diff
@@ -0,0 +1,12 @@
+Clarke Littlejohn
+
+
+1) No the code is not valid do to the printf(). it should be printf("%d",somenumber);. If that is fixed you can assume that the stdio.h is included, 
+that it is in the main function, that i is initialized before the for loop, and that it returns 0 then the code should work.
+
+2) the code should work but it is useless. it is and infinite loop the rror in this might be that the i-- should be i++
+
+3)i   O(n^2)
+  ii  O(n^3)
```

Comment:
The answer is actually O(n^4) - again, for loops that run from 0 to n have the time complexity O(n). So the top loop is O(n), the middle Loop is O(n^2) iterations because it iterates from 1 to i^2. The if statement is O(1 / n) times because of the modulo function. Lastly, the inner loop  is O(n^2) iterations because it iterates from 0 to j and j = i^2 = n^2. 

Therefore, the sum = -(1/12)*n + (3/8)*n^2 - (5/12)*n^3 + (1/8)*n^4  = O(n^4)